### PR TITLE
Migrate to MSK

### PR DESCRIPTION
**Pull request by elenasuteva**
# Description

## What

Using the Kafka-client-go/v3 library replaced the consumer of Kafka messages;
Updated env vars;
Replace logrus with upp logger v2;
Updated go version to 18;
Fixed tests.

## Why

https://financialtimes.atlassian.net/browse/UPPSF-3491

## Anything, in particular, you'd like to highlight to reviewers

Mention here sections of code which you would like reviewers to pay extra attention to .E.g

_Would appreciate a second pair of eyes on the test_  
_I am not quite sure how this bit works_  
_Is there a better library for doing x_  

## Scope and particulars of this PR (Please tick all that apply)

- [ ] Tech hygiene (dependency updating & other tech debt)
- [ ] Bug fix
- [x] Feature
- [ ] Documentation
- [ ] Breaking change
- [ ] Minor change (e.g. fixing a typo, adding config)

___
This Pull Request follows the rules described in our [Pull Requests Guide](https://github.com/Financial-Times/upp-docs/tree/master/guides/pr-guide)

### CODE REVIEW

File: config.json.template [hunk 1]
```diff
@@ -2,12 +2,12 @@
   "threshold": 120,
   "queueConfig": {
     "address": [
```
Comment (atanasdinov) [inline config.json.template:4]: Is this now obsolete with the addition of `connectionString`?

File: config/config.go [hunk 2]
```diff
@@ -5,22 +5,30 @@ import (
...
+	GraphiteAddress     string            `json:"graphiteAddress"`
+	GraphiteUUID        string            `json:"graphiteUUID"`
+	Environment         string            `json:"environment"`
+}
+
+// QueueConfig is the configuration for kafka consumer queue
+type QueueConfig struct {
+	Address                 []string `json:"address"`
+	ConsumerGroup           string   `json:"consumerGroup"`
+	KafkaLagTolerance       string   `json:"lagTolerance"`
```
Comment (atanasdinov) [inline config/config.go:29]: Is there a reason for not using `int` directly? We need to manually convert this later.

File: README.md [hunk 3]
```diff
@@ -45,11 +45,11 @@ __Note that deployment to FTP2 is no longer supported.__
 	//URL(s) of the messaging system
 	"address": [queue address1, queue address 2],
 	//The group name by which this app will be knows to the messaging system
-	"group": "YourGroupName",
+	"ConsumerGroup": "YourGroupName",
```
Comment (atanasdinov) [inline README.md:48]: This being in PascalCase doesn't really make sense to me? Is it tested?

File: main.go [hunk 4]
```diff
@@ -113,27 +122,24 @@ func loadHistory(metricContainer *metrics.History) func(w http.ResponseWriter, r
...
+func readKafkaMessages(c *kafka.Consumer, h MessageHandler, environments *envs.Environments, log *logger.UPPLogger) {
 	for !environments.AreReady() {
 		log.Info("Environments not set, retry in 3s...")
 		time.Sleep(3 * time.Second)
 	}
 
-	h := NewKafkaMessageHandler(appConfig, environments, subscribedFeeds, metricSink, metricContainer, e2eTestUUIDs)
-	c := consumer.NewConsumer(appConfig.QueueConf, h.HandleMessage, &http.Client{})
-
 	var wg sync.WaitGroup
```
Comment (atanasdinov) [inline main.go:131]: A wait group is not needed anymore.

**Review by atanasdinov — CHANGES_REQUESTED** [review]

**Comment by atanasdinov** [comment]
Also the PR states that `logrus` is replaced but in fact it's used in many places. It's still probably not the best idea to change those usages in this PR tho.

File: config/config.go (hunk 2 above)
Comment (elenasuteva) [reply config/config.go:29]: The reason is that JSON unmarshalling was failing on dev. I am going to inspect this once again and try to change it to int. I think that the reason for failing is config.json parses "" for kafkaLagTolerance if not set correctly.

Comment (atanasdinov) [reply config/config.go:29]: I suppose that issue had been caused by the value being string in the JSON itself instead of integer.

Comment (elenasuteva) [reply config/config.go]: I've fixed it now I think and tested it.

File: checks/preCheck.go [hunk 5]
```diff
@@ -7,7 +7,7 @@ import (
 	"github.com/Financial-Times/publish-availability-monitor/content"
 	"github.com/Financial-Times/publish-availability-monitor/envs"
 	"github.com/Financial-Times/publish-availability-monitor/metrics"
-	log "github.com/Sirupsen/logrus"
+	log "github.com/sirupsen/logrus"
```
Comment (ilian2233) [inline checks/preCheck.go:10]: Shouldn't we replace `github.com/sirupsen/logrus` with `github.com/Financial-Times/go-logger/v2`

File: go.mod [hunk 6]
```diff
@@ -1,24 +1,44 @@
 module github.com/Financial-Times/publish-availability-monitor
 
-go 1.13
+go 1.18
```
Comment (ilian2233) [inline go.mod:3]: **nitpick (non-blocking):** can be updated to 1.19

File: logformat/logFormatter.go [hunk 7]
```diff
@@ -5,7 +5,7 @@ import (
 	"runtime"
```
Comment (ilian2233) [inline logformat/logFormatter.go:5]: **question (non-blocking):** Is that file ever used?

Maybe should be removed?

**Review by martin-stanchev — APPROVED** [review]

**Review by martin-stanchev — APPROVED** [review]

**Review by dtvalk-ov — APPROVED** [review]

**Review by atanasdinov — APPROVED** [review]

**Pull request merged**